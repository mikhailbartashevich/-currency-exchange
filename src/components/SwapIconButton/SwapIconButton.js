import React from 'react';
import { withStyles } from '@material-ui/styles';
import { pure } from 'recompose';
import { Icon, IconButton } from '@material-ui/core';

const styles = {
  root: {
    color: '#446dd2',
  },
};

const SwapIconButtonRaw = props => {
  const { classes, color, ...other } = props;
  return (
    <IconButton size="small" {...other} className="SwapIconButton">
      <Icon className={classes.root}>
        swap_vert
      </Icon>
    </IconButton>
  );
};

export const SwapIconButton = pure(withStyles(styles)(SwapIconButtonRaw));
