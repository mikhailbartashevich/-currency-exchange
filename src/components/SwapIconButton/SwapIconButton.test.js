import React from 'react';
import { shallow } from 'enzyme';
import { SwapIconButton } from './SwapIconButton';

describe('SwapIconButton', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<SwapIconButton debug />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with no props', () => {
    const component = shallow(<SwapIconButton />);
    expect(component).toMatchSnapshot();
  });

});
