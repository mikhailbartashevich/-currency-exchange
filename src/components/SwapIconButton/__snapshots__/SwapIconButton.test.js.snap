// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapIconButton should render correctly in "debug" mode 1`] = `
<WithStyles(SwapIconButtonRaw)
  debug={true}
/>
`;

exports[`SwapIconButton should render correctly with no props 1`] = `<WithStyles(SwapIconButtonRaw) />`;
